### PR TITLE
BoardGame Fix & Interaction Change

### PR DESCRIPTION
**Pull request by EgardA**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behaviour?

Tablut works fine locally but other board games remain untested but not used in current vertical slice tavern anyway.

## What is the new behaviour?

Resolves #

## Other information

### CODE REVIEW

File: source/Common - Copy/Common.csproj [hunk 1]
```diff
@@ -0,0 +1,119 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (smudge202) [inline source/Common - Copy/Common.csproj]: This class (and everything in this directory) was in a branch that Joke rejected: https://github.com/Bannerlord-Coop-Team/BannerlordCoop/pull/438
You'll essentially want to revert this commit on your branch: https://github.com/Bannerlord-Coop-Team/BannerlordCoop/pull/438/commits/0ab2608ef3761917187fab55cf193e15a77f2be1

File: source/Missions/BoardGameLogic.cs [hunk 2]
```diff
@@ -103,40 +106,30 @@ private void PreplaceUnits()
...
 
-        private void SeegaPawnCapture(PawnBase pawn)
+        private void OnPawnCapture(PawnBase pawn)
         {
             //Only call SetPawnCaptured when it's a forceful remove as a result of no moves available as otherwise it gets handled locally from the move
-            if (!FocusBlockingPawnsPatch.ForceRemove)
-            {
-                return;
-            }
+            //if (!FocusBlockingPawnsPatch.ForceRemove)
```
Comment (smudge202) [inline source/Missions/BoardGameLogic.cs:112]: nit: commented out code should be deleted?

File: source/Missions/Patch/Agents/AgentInteractionPatch.cs [hunk 3]
```diff
@@ -2,26 +2,51 @@
...
+            return true;
 
-            // TODO fix board games
-            //OnAgentInteraction?.Invoke(userAgent, agent);
+        }
+    }
 
-            return false;
+    [HarmonyPatch(typeof(ConversationManager), "ProcessSentence")]
+    public class ProcessSentencePatch
```
Comment (smudge202) [inline source/Missions/Patch/Agents/AgentInteractionPatch.cs]: Should this be in it's own file like the other patches? Or maybe the file renamed to be `AgentInteractionPatches` like the BoardGame ones?

File: source/Missions/Patch/BoardGames/BoardGamePatches.cs [hunk 4]
```diff
@@ -30,40 +33,20 @@ static void Postfix(Agent ownerAgent)
...
+        private static readonly PropertyInfo AgentNavigatorPropertyInfo = typeof(CampaignAgentComponent).GetProperty("AgentNavigator");
         public static event Action<MissionBoardGameLogic> OnGameOver;
         static bool Prefix(MissionBoardGameLogic __instance, Agent conversationAgent)
         {
             if (NetworkAgentRegistry.AgentToId.ContainsKey(conversationAgent))
             {
                 OnGameOver?.Invoke(__instance);
 
+                //Set AgentNavigator to null as this gets set in SetGameOver by default and destroys all future interactions
+                //PropertyInfo prop = conversationAgent.GetComponent<CampaignAgentComponent>().GetType().GetProperty("AgentNavigator", BindingFlags.Public | BindingFlags.Instance);
```
Comment (smudge202) [inline source/Missions/Patch/BoardGames/BoardGamePatches.cs]: nit: commented out code

File: source/Missions/Patch/BoardGames/BoardGamePatches.cs [hunk 5]
```diff
@@ -119,13 +99,14 @@ static bool Postfix(bool result)
...
-    public class HandlePlayerInputPatch
+    [HarmonyPatch(typeof(BoardGameAIBase), "CalculateMovementStageMoveOnSeparateThread")]
+    public class CalculateMovePatch
     {
-        public static event Action<Move> OnHandlePlayerInput;
-        static void Postfix(ref BoardGameBase __instance, ref Move __result)
+        public static bool Prefix()
         {
-            OnHandlePlayerInput?.Invoke(__result);
+            if (!BoardGameLogic.IsPlayingOtherPlayer) return true;
```
Comment (smudge202) [inline source/Missions/Patch/BoardGames/BoardGamePatches.cs]: nit: no need for an if statement, just `return BoardGameLogic.IsPlayingOtherPlayer`?

**Review by smudge202 — CHANGES_REQUESTED** [review]

File: source/Missions/BoardGameLogic.cs (hunk 2 above)
Comment (EgardA) [reply source/Missions/BoardGameLogic.cs:112]: Want to keep these comments as I will need them almost certainly when fixing the other board games, particularly Seega

**Review by smudge202 — APPROVED** [review]

**Pull request merged**